"pkg-config --exists sqlite3"
| pkg-config --libs sqlite3
=> "-lsqlite3  \n"
"gcc -o conftest -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.    -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -mtune=generic -fPIC conftest.c  -L. -L/usr/lib64 -L. -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector -rdynamic -Wl,-export-dynamic  -m64   -lruby  -lpthread -ldl -lcrypt -lm   -lc"
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: int main(int argc, char **argv)
4: {
5:   return 0;
6: }
/* end */

"gcc -o conftest -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.    -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -mtune=generic -fPIC conftest.c  -L. -L/usr/lib64 -L. -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector -rdynamic -Wl,-export-dynamic  -m64   -lruby -lsqlite3 -lpthread -ldl -lcrypt -lm   -lc"
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: int main(int argc, char **argv)
4: {
5:   return 0;
6: }
/* end */

| pkg-config --cflags-only-I sqlite3
=> " \n"
| pkg-config --cflags-only-other sqlite3
=> " \n"
| pkg-config --libs-only-l sqlite3
=> "-lsqlite3  \n"
package configuration for sqlite3
cflags: 
ldflags: 
libs: -lsqlite3

find_header: checking for sqlite3.h... -------------------- yes

"gcc -E -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.     -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -mtune=generic -fPIC  -DTAINTING_SUPPORT  conftest.c -o conftest.i"
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <sqlite3.h>
/* end */

--------------------

find_library: checking for pthread_create() in -lpthread... -------------------- yes

"gcc -o conftest -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.     -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -mtune=generic -fPIC  -DTAINTING_SUPPORT conftest.c  -L. -L/usr/lib64 -L. -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector -rdynamic -Wl,-export-dynamic   -m64   -lsqlite3 -lruby -lpthread  -lsqlite3 -lpthread -ldl -lcrypt -lm   -lc"
conftest.c: 関数 ‘t’ 内:
conftest.c:13:57: エラー: ‘pthread_create’ undeclared (first use in this function); did you mean ‘rb_thread_create’?
 int t(void) { void ((*volatile p)()); p = (void ((*)()))pthread_create; return !p; }
                                                         ^~~~~~~~~~~~~~
                                                         rb_thread_create
conftest.c:13:57: 備考: 未宣言の識別子は出現した各関数内で一回のみ報告されます
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     printf("%p", &t);
 9:   }
10: 
11:   return 0;
12: }
13: int t(void) { void ((*volatile p)()); p = (void ((*)()))pthread_create; return !p; }
/* end */

"gcc -o conftest -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.     -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -mtune=generic -fPIC  -DTAINTING_SUPPORT conftest.c  -L. -L/usr/lib64 -L. -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector -rdynamic -Wl,-export-dynamic   -m64   -lsqlite3 -lruby -lpthread  -lsqlite3 -lpthread -ldl -lcrypt -lm   -lc"
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     printf("%p", &t);
 9:   }
10: 
11:   return 0;
12: }
13: extern void pthread_create();
14: int t(void) { pthread_create(); return 0; }
/* end */

--------------------

have_library: checking for -ldl... -------------------- yes

"gcc -o conftest -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.     -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -mtune=generic -fPIC  -DTAINTING_SUPPORT conftest.c  -L. -L/usr/lib64 -L. -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector -rdynamic -Wl,-export-dynamic   -m64  -lpthread  -lsqlite3 -lruby -ldl -lpthread  -lsqlite3 -lpthread -ldl -lcrypt -lm   -lc"
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     printf("%p", &t);
 9:   }
10: 
11:   return 0;
12: }
13: 
14: int t(void) { ; return 0; }
/* end */

--------------------

find_library: checking for sqlite3_libversion_number() in -lsqlite3... -------------------- yes

"gcc -o conftest -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.     -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -mtune=generic -fPIC  -DTAINTING_SUPPORT conftest.c  -L. -L/usr/lib64 -L. -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector -rdynamic -Wl,-export-dynamic   -m64  -ldl -lpthread  -lsqlite3 -lruby -lsqlite3 -ldl -lpthread  -lsqlite3 -lpthread -ldl -lcrypt -lm   -lc"
conftest.c: 関数 ‘t’ 内:
conftest.c:13:57: エラー: ‘sqlite3_libversion_number’ が宣言されていません (この関数内での最初の使用)
 int t(void) { void ((*volatile p)()); p = (void ((*)()))sqlite3_libversion_number; return !p; }
                                                         ^~~~~~~~~~~~~~~~~~~~~~~~~
conftest.c:13:57: 備考: 未宣言の識別子は出現した各関数内で一回のみ報告されます
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     printf("%p", &t);
 9:   }
10: 
11:   return 0;
12: }
13: int t(void) { void ((*volatile p)()); p = (void ((*)()))sqlite3_libversion_number; return !p; }
/* end */

"gcc -o conftest -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.     -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -mtune=generic -fPIC  -DTAINTING_SUPPORT conftest.c  -L. -L/usr/lib64 -L. -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector -rdynamic -Wl,-export-dynamic   -m64  -ldl -lpthread  -lsqlite3 -lruby -lsqlite3 -ldl -lpthread  -lsqlite3 -lpthread -ldl -lcrypt -lm   -lc"
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     printf("%p", &t);
 9:   }
10: 
11:   return 0;
12: }
13: extern void sqlite3_libversion_number();
14: int t(void) { sqlite3_libversion_number(); return 0; }
/* end */

--------------------

have_func: checking for rb_proc_arity()... -------------------- yes

"gcc -o conftest -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.     -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -mtune=generic -fPIC  -DTAINTING_SUPPORT conftest.c  -L. -L/usr/lib64 -L. -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector -rdynamic -Wl,-export-dynamic   -m64  -lsqlite3 -ldl -lpthread  -lsqlite3 -lruby -lsqlite3 -ldl -lpthread  -lsqlite3 -lpthread -ldl -lcrypt -lm   -lc"
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     printf("%p", &t);
 9:   }
10: 
11:   return 0;
12: }
13: int t(void) { void ((*volatile p)()); p = (void ((*)()))rb_proc_arity; return !p; }
/* end */

--------------------

have_func: checking for rb_integer_pack()... -------------------- yes

"gcc -o conftest -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.     -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -mtune=generic -fPIC  -DTAINTING_SUPPORT conftest.c  -L. -L/usr/lib64 -L. -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector -rdynamic -Wl,-export-dynamic   -m64  -lsqlite3 -ldl -lpthread  -lsqlite3 -lruby -lsqlite3 -ldl -lpthread  -lsqlite3 -lpthread -ldl -lcrypt -lm   -lc"
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     printf("%p", &t);
 9:   }
10: 
11:   return 0;
12: }
13: int t(void) { void ((*volatile p)()); p = (void ((*)()))rb_integer_pack; return !p; }
/* end */

--------------------

have_func: checking for sqlite3_initialize()... -------------------- yes

"gcc -o conftest -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.     -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -mtune=generic -fPIC  -DTAINTING_SUPPORT conftest.c  -L. -L/usr/lib64 -L. -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector -rdynamic -Wl,-export-dynamic   -m64  -lsqlite3 -ldl -lpthread  -lsqlite3 -lruby -lsqlite3 -ldl -lpthread  -lsqlite3 -lpthread -ldl -lcrypt -lm   -lc"
conftest.c: 関数 ‘t’ 内:
conftest.c:13:57: エラー: ‘sqlite3_initialize’ undeclared (first use in this function); did you mean ‘rb_struct_initialize’?
 int t(void) { void ((*volatile p)()); p = (void ((*)()))sqlite3_initialize; return !p; }
                                                         ^~~~~~~~~~~~~~~~~~
                                                         rb_struct_initialize
conftest.c:13:57: 備考: 未宣言の識別子は出現した各関数内で一回のみ報告されます
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     printf("%p", &t);
 9:   }
10: 
11:   return 0;
12: }
13: int t(void) { void ((*volatile p)()); p = (void ((*)()))sqlite3_initialize; return !p; }
/* end */

"gcc -o conftest -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.     -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -mtune=generic -fPIC  -DTAINTING_SUPPORT conftest.c  -L. -L/usr/lib64 -L. -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector -rdynamic -Wl,-export-dynamic   -m64  -lsqlite3 -ldl -lpthread  -lsqlite3 -lruby -lsqlite3 -ldl -lpthread  -lsqlite3 -lpthread -ldl -lcrypt -lm   -lc"
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     printf("%p", &t);
 9:   }
10: 
11:   return 0;
12: }
13: extern void sqlite3_initialize();
14: int t(void) { sqlite3_initialize(); return 0; }
/* end */

--------------------

have_func: checking for sqlite3_backup_init()... -------------------- yes

"gcc -o conftest -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.     -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -mtune=generic -fPIC  -DTAINTING_SUPPORT conftest.c  -L. -L/usr/lib64 -L. -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector -rdynamic -Wl,-export-dynamic   -m64  -lsqlite3 -ldl -lpthread  -lsqlite3 -lruby -lsqlite3 -ldl -lpthread  -lsqlite3 -lpthread -ldl -lcrypt -lm   -lc"
conftest.c: 関数 ‘t’ 内:
conftest.c:13:57: エラー: ‘sqlite3_backup_init’ が宣言されていません (この関数内での最初の使用)
 int t(void) { void ((*volatile p)()); p = (void ((*)()))sqlite3_backup_init; return !p; }
                                                         ^~~~~~~~~~~~~~~~~~~
conftest.c:13:57: 備考: 未宣言の識別子は出現した各関数内で一回のみ報告されます
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     printf("%p", &t);
 9:   }
10: 
11:   return 0;
12: }
13: int t(void) { void ((*volatile p)()); p = (void ((*)()))sqlite3_backup_init; return !p; }
/* end */

"gcc -o conftest -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.     -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -mtune=generic -fPIC  -DTAINTING_SUPPORT conftest.c  -L. -L/usr/lib64 -L. -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector -rdynamic -Wl,-export-dynamic   -m64  -lsqlite3 -ldl -lpthread  -lsqlite3 -lruby -lsqlite3 -ldl -lpthread  -lsqlite3 -lpthread -ldl -lcrypt -lm   -lc"
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     printf("%p", &t);
 9:   }
10: 
11:   return 0;
12: }
13: extern void sqlite3_backup_init();
14: int t(void) { sqlite3_backup_init(); return 0; }
/* end */

--------------------

have_func: checking for sqlite3_column_database_name()... -------------------- yes

"gcc -o conftest -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.     -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -mtune=generic -fPIC  -DTAINTING_SUPPORT conftest.c  -L. -L/usr/lib64 -L. -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector -rdynamic -Wl,-export-dynamic   -m64  -lsqlite3 -ldl -lpthread  -lsqlite3 -lruby -lsqlite3 -ldl -lpthread  -lsqlite3 -lpthread -ldl -lcrypt -lm   -lc"
conftest.c: 関数 ‘t’ 内:
conftest.c:13:57: エラー: ‘sqlite3_column_database_name’ が宣言されていません (この関数内での最初の使用)
 int t(void) { void ((*volatile p)()); p = (void ((*)()))sqlite3_column_database_name; return !p; }
                                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
conftest.c:13:57: 備考: 未宣言の識別子は出現した各関数内で一回のみ報告されます
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     printf("%p", &t);
 9:   }
10: 
11:   return 0;
12: }
13: int t(void) { void ((*volatile p)()); p = (void ((*)()))sqlite3_column_database_name; return !p; }
/* end */

"gcc -o conftest -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.     -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -mtune=generic -fPIC  -DTAINTING_SUPPORT conftest.c  -L. -L/usr/lib64 -L. -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector -rdynamic -Wl,-export-dynamic   -m64  -lsqlite3 -ldl -lpthread  -lsqlite3 -lruby -lsqlite3 -ldl -lpthread  -lsqlite3 -lpthread -ldl -lcrypt -lm   -lc"
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     printf("%p", &t);
 9:   }
10: 
11:   return 0;
12: }
13: extern void sqlite3_column_database_name();
14: int t(void) { sqlite3_column_database_name(); return 0; }
/* end */

--------------------

have_func: checking for sqlite3_enable_load_extension()... -------------------- yes

"gcc -o conftest -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.     -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -mtune=generic -fPIC  -DTAINTING_SUPPORT conftest.c  -L. -L/usr/lib64 -L. -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector -rdynamic -Wl,-export-dynamic   -m64  -lsqlite3 -ldl -lpthread  -lsqlite3 -lruby -lsqlite3 -ldl -lpthread  -lsqlite3 -lpthread -ldl -lcrypt -lm   -lc"
conftest.c: 関数 ‘t’ 内:
conftest.c:13:57: エラー: ‘sqlite3_enable_load_extension’ undeclared (first use in this function); did you mean ‘__glibc_clang_has_extension’?
 int t(void) { void ((*volatile p)()); p = (void ((*)()))sqlite3_enable_load_extension; return !p; }
                                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                                                         __glibc_clang_has_extension
conftest.c:13:57: 備考: 未宣言の識別子は出現した各関数内で一回のみ報告されます
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     printf("%p", &t);
 9:   }
10: 
11:   return 0;
12: }
13: int t(void) { void ((*volatile p)()); p = (void ((*)()))sqlite3_enable_load_extension; return !p; }
/* end */

"gcc -o conftest -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.     -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -mtune=generic -fPIC  -DTAINTING_SUPPORT conftest.c  -L. -L/usr/lib64 -L. -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector -rdynamic -Wl,-export-dynamic   -m64  -lsqlite3 -ldl -lpthread  -lsqlite3 -lruby -lsqlite3 -ldl -lpthread  -lsqlite3 -lpthread -ldl -lcrypt -lm   -lc"
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     printf("%p", &t);
 9:   }
10: 
11:   return 0;
12: }
13: extern void sqlite3_enable_load_extension();
14: int t(void) { sqlite3_enable_load_extension(); return 0; }
/* end */

--------------------

have_func: checking for sqlite3_load_extension()... -------------------- yes

"gcc -o conftest -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.     -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -mtune=generic -fPIC  -DTAINTING_SUPPORT conftest.c  -L. -L/usr/lib64 -L. -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector -rdynamic -Wl,-export-dynamic   -m64  -lsqlite3 -ldl -lpthread  -lsqlite3 -lruby -lsqlite3 -ldl -lpthread  -lsqlite3 -lpthread -ldl -lcrypt -lm   -lc"
conftest.c: 関数 ‘t’ 内:
conftest.c:13:57: エラー: ‘sqlite3_load_extension’ undeclared (first use in this function); did you mean ‘__glibc_clang_has_extension’?
 int t(void) { void ((*volatile p)()); p = (void ((*)()))sqlite3_load_extension; return !p; }
                                                         ^~~~~~~~~~~~~~~~~~~~~~
                                                         __glibc_clang_has_extension
conftest.c:13:57: 備考: 未宣言の識別子は出現した各関数内で一回のみ報告されます
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     printf("%p", &t);
 9:   }
10: 
11:   return 0;
12: }
13: int t(void) { void ((*volatile p)()); p = (void ((*)()))sqlite3_load_extension; return !p; }
/* end */

"gcc -o conftest -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.     -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -mtune=generic -fPIC  -DTAINTING_SUPPORT conftest.c  -L. -L/usr/lib64 -L. -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector -rdynamic -Wl,-export-dynamic   -m64  -lsqlite3 -ldl -lpthread  -lsqlite3 -lruby -lsqlite3 -ldl -lpthread  -lsqlite3 -lpthread -ldl -lcrypt -lm   -lc"
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     printf("%p", &t);
 9:   }
10: 
11:   return 0;
12: }
13: extern void sqlite3_load_extension();
14: int t(void) { sqlite3_load_extension(); return 0; }
/* end */

--------------------

have_func: checking for sqlite3_open_v2()... -------------------- yes

"gcc -o conftest -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.     -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -mtune=generic -fPIC  -DTAINTING_SUPPORT conftest.c  -L. -L/usr/lib64 -L. -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector -rdynamic -Wl,-export-dynamic   -m64  -lsqlite3 -ldl -lpthread  -lsqlite3 -lruby -lsqlite3 -ldl -lpthread  -lsqlite3 -lpthread -ldl -lcrypt -lm   -lc"
conftest.c: 関数 ‘t’ 内:
conftest.c:13:57: エラー: ‘sqlite3_open_v2’ が宣言されていません (この関数内での最初の使用)
 int t(void) { void ((*volatile p)()); p = (void ((*)()))sqlite3_open_v2; return !p; }
                                                         ^~~~~~~~~~~~~~~
conftest.c:13:57: 備考: 未宣言の識別子は出現した各関数内で一回のみ報告されます
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     printf("%p", &t);
 9:   }
10: 
11:   return 0;
12: }
13: int t(void) { void ((*volatile p)()); p = (void ((*)()))sqlite3_open_v2; return !p; }
/* end */

"gcc -o conftest -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.     -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -mtune=generic -fPIC  -DTAINTING_SUPPORT conftest.c  -L. -L/usr/lib64 -L. -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector -rdynamic -Wl,-export-dynamic   -m64  -lsqlite3 -ldl -lpthread  -lsqlite3 -lruby -lsqlite3 -ldl -lpthread  -lsqlite3 -lpthread -ldl -lcrypt -lm   -lc"
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     printf("%p", &t);
 9:   }
10: 
11:   return 0;
12: }
13: extern void sqlite3_open_v2();
14: int t(void) { sqlite3_open_v2(); return 0; }
/* end */

--------------------

have_func: checking for sqlite3_prepare_v2()... -------------------- yes

"gcc -o conftest -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.     -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -mtune=generic -fPIC  -DTAINTING_SUPPORT conftest.c  -L. -L/usr/lib64 -L. -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector -rdynamic -Wl,-export-dynamic   -m64  -lsqlite3 -ldl -lpthread  -lsqlite3 -lruby -lsqlite3 -ldl -lpthread  -lsqlite3 -lpthread -ldl -lcrypt -lm   -lc"
conftest.c: 関数 ‘t’ 内:
conftest.c:13:57: エラー: ‘sqlite3_prepare_v2’ が宣言されていません (この関数内での最初の使用)
 int t(void) { void ((*volatile p)()); p = (void ((*)()))sqlite3_prepare_v2; return !p; }
                                                         ^~~~~~~~~~~~~~~~~~
conftest.c:13:57: 備考: 未宣言の識別子は出現した各関数内で一回のみ報告されます
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     printf("%p", &t);
 9:   }
10: 
11:   return 0;
12: }
13: int t(void) { void ((*volatile p)()); p = (void ((*)()))sqlite3_prepare_v2; return !p; }
/* end */

"gcc -o conftest -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.     -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -mtune=generic -fPIC  -DTAINTING_SUPPORT conftest.c  -L. -L/usr/lib64 -L. -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector -rdynamic -Wl,-export-dynamic   -m64  -lsqlite3 -ldl -lpthread  -lsqlite3 -lruby -lsqlite3 -ldl -lpthread  -lsqlite3 -lpthread -ldl -lcrypt -lm   -lc"
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     printf("%p", &t);
 9:   }
10: 
11:   return 0;
12: }
13: extern void sqlite3_prepare_v2();
14: int t(void) { sqlite3_prepare_v2(); return 0; }
/* end */

--------------------

have_type: checking for sqlite3_int64 in sqlite3.h... -------------------- yes

"gcc -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.     -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -mtune=generic -fPIC  -DTAINTING_SUPPORT -m64  -c conftest.c"
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <sqlite3.h>
4: 
5: /*top*/
6: typedef sqlite3_int64 conftest_type;
7: int conftestval[sizeof(conftest_type)?1:-1];
/* end */

--------------------

have_type: checking for sqlite3_uint64 in sqlite3.h... -------------------- yes

"gcc -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.     -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -mtune=generic -fPIC  -DTAINTING_SUPPORT -m64  -c conftest.c"
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <sqlite3.h>
4: 
5: /*top*/
6: typedef sqlite3_uint64 conftest_type;
7: int conftestval[sizeof(conftest_type)?1:-1];
/* end */

--------------------

